src/OPTokenClaim.sol

```diff
@@ -54,62 +60,107 @@ contract OPTokenClaim is Ownable {
         emit ClaimExtended(months);
     }
 
-    function claimOP(address account) external {
+    /* ========== VIEWS ========== */
+
+    /// @return epochNumber the current epoch number
+    function currentEpoch() external view returns (uint256 epochNumber) {
+        return _currentEpoch(config);
+    }
+
+    /// returns last epoch where claims are open
+    function maxEpoch() external view returns (uint256) {
+        return config.maxEpoch;
+    }
+
+    /// @return reward the amount of OP tokens to be claimed
+    function calcReward(address account, uint256 epochNum) public view returns (uint256 reward) {
+        unchecked {
+            // calculate the total reward of given epoch
+            uint256 totalReward = totalEXPAtEpoch[epochNum] * 5;
+
+            // calculate individual reward
+            uint256 subscribedEXP = epochToSubscribedEXP[epochNum][account];
+            if (totalReward > MAX_REWARD) {
+                reward = 5 * subscribedEXP * MAX_REWARD / totalReward;
+            } else {
+                reward = 5 * subscribedEXP;
+            }
```

Comment:
if totalReward == 95
user subscribed with 99 EXP
the reward to the user will be 5*99, is that ok?
Can we be in this situation?

Reply:
do you mean `totalReward == 495`?

if a user subscribed with 99 EXP total reward is 495, that means no other user subscribed. So in that case the user would just get his share normally, nothing weird

Reply:
total reward can never be less than 495? i did not do the math yet

Reply:
`totalReward` is `5 * totalEXP`

`totalEXP` is the sum of all `subscribedEXP`, so this should be true for all accounts:

    totalEXP >= subscribedEXP(account)